tests/layering/layering_test.go

```diff
@@ -0,0 +1,148 @@
+package e2e_test
+
+import (
+	"context"
+	"flag"
+	"testing"
+
+	"github.com/openshift/machine-config-operator/test/framework"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var deleteBuild = flag.Bool("delete-build", false, "Delete the derived build at the end of the test.")
+var deleteMachine = flag.Bool("delete-machine", false, "Delete the target machine after test run.")
+var streamBuild = flag.Bool("stream-build", false, "Stream the derived image build to stdout")
+var buildLogFile = flag.String("build-log", "", "Filename to write the derived image build log to")
+
+const (
+	imageStreamName = "test-boot-in-cluster-image"
+	buildName       = imageStreamName
+	// ostreeUnverifiedRegistry means no GPG or container signatures are used.
+	// Right now we're usually relying on digested pulls. See
+	// https://github.com/openshift/machine-config-operator/blob/master/docs/OSUpgrades.md#questions-and-answers around integrity.
+	// See https://docs.rs/ostree-ext/0.5.1/ostree_ext/container/struct.OstreeImageReference.html
+	ostreeUnverifiedRegistry = "ostree-unverified-registry"
+	imageRegistry            = "image-registry.openshift-image-registry.svc:5000"
+	// If this moves from /run, make sure files get cleaned up
+	authfilePath   = "/run/ostree/auth.json"
+	mcoNamespace   = "openshift-machine-config-operator"
+	imagePullSpec  = "registry.ci.openshift.org/rhcos-devel/rhel-coreos:%s"
```

Comment:
q: why %s here?

Reply:
This value gets replaced in with either `latest` or the OCP release (e.g., `4.11`) using https://github.com/openshift/os/pull/746/files#diff-58ffaa9005e1e047a4897d0a0d8a414238a5334c730449caa62fbd8fca7cb34dR36.

This occurs here: https://github.com/openshift/os/pull/746/files#diff-58ffaa9005e1e047a4897d0a0d8a414238a5334c730449caa62fbd8fca7cb34dR75.

My reasoning was that this could run on all of the release branches as well as `master`, so it should pull the appropriate image for that branch.

Reply:
for some reason link not working what line is it (just for my own info when you get a chance)

Reply:
tests/layering/helpers_test.go:75